### PR TITLE
[DO NOT MERGE] refactor: use widgetParams from renderer

### PR DESCRIPTION
**Pull request by samouss**
**Summary**

This is an example for the simplification of the widget renderer (see: #2473).

### CODE REVIEW

File: src/widgets/range-slider/range-slider.js [hunk 1]
```diff
@@ -162,47 +167,35 @@ rangeSlider({
...
-    cssClasses,
-  });
+  if (!widgetParams.container) {
+    throw new Error(usage);
+  }
 
   try {
-    const makeWidget = connectRangeSlider(specializedRenderer);
-    return makeWidget({ attributeName, min, max, precision });
+    const makeWidget = connectRangeSlider(renderer);
```
Comment (bobylito) [inline src/widgets/range-slider/range-slider.js]: The nice thing is that we can even move that out: having a single instance is possible.

File: src/widgets/range-slider/range-slider.js [hunk 2]
```diff
@@ -162,47 +167,35 @@ rangeSlider({
...
+  const widgetParams = {
+    container: '',
+    step: 1,
+    pips: true,
+    tooltips: true,
+    autoHideContainer: true,
+    cssClasses: {},
+    renderState: {},
+    templates: defaultTemplates,
+    ...params,
```
Comment (bobylito) [inline src/widgets/range-slider/range-slider.js]: Someone can pass a new renderState here.

**Comment by bobylito** [comment]
I think it could really simplifies thinking about the code in the widgets. It works for me because the connect(renderer) that we use will be hidden which is what we want because we have some plumbing that we want to hide (renderState).

File: src/widgets/range-slider/range-slider.js (hunk 2 above)
Comment (samouss) [reply src/widgets/range-slider/range-slider.js]: Yep, you are right! But it could be hide, I will update it.

**Comment by bobylito** [comment]
Imho that's a great improvement over the current pattern for implementing widget in InstantSearch.js. We should definetely move forward with it. However this is not high priority as we haven't identified a bug with the current approach. Are you ok @samouss for moving that to 2018?

**Comment by iam4x** [comment]
I agree with @bobylito it would be great to don't move the connectors API until we have a v1 of angular-instantsearch 👍

**Comment by samouss** [comment]
@bobylito No problem, it was just a proposition. I close the PR but I leave the issue open for later.

**Comment by bobylito** [comment]
> @bobylito No problem, it was just a proposition. I close the PR but I leave the issue open for later.

And just to be perfectly clear: I love it :)

**Pull request closed without merging**